### ISSUE TITLE
Publicize: removing user from site doesn't remove Publicize connection available to other users

### ISSUE THREAD

**Issue by liviopv** [Automattic/jetpack#18434]
#### Steps to reproduce the issue

1. From User Account A, add a new user account to the site from My Site > Manage > People, which will call User Account B
2. Accept the invitation for User Account B
3. Log in as User Account B
4. Connect a new social account as User Account B and select `Connection available to all administrators, editors, and authors` 
5. As User Account A, remove User Account B from the site
6. Start a new post and check the sharing settings: the social account from User Account B (no longer a user on the site) will still be available as a sharing option

#### What I expected
When a user is deleted from a site, all Publicize connections created by that user should be removed.

#### What happened instead
The social connection from the deleted user was still available to other users as a sharing option.

User-report: #3637917-zen

Marking as "high" for the possible privacy concern.

**Comment by danmaze** [comment]
I'm not able to replicate this, so I'm guessing it was fixed in a follow-up release. 

> Connect a new social account as User Account B and select Connection available to all administrators, editors, and authors

I didn't see the option to make the connection for user account B global. I don't suppose that's because the user I added to the site was an author and did not have administrative privileges.

> Start a new post and check the sharing settings: the social account from User Account B (no longer a user on the site) will still be available as a sharing option

After deleting user account B from the site, I didn't see their connection on the "Tools > Marketing > Connections" screen and when I create a new post for sharing.
